project/src/main/webapp/script.js

```diff
@@ -62,11 +121,131 @@ function createPlaceElement(place) {
 }
 
 /**
- * Displays the query form to the user and hides the results, so that the user can try again with a
- * different query.
+ * Displays the query form to the user and hides the results, so that the user can try to get new
+ * results.
  */
 function tryAgain() {
   document.getElementById('query-form').style.display = 'block';
   document.getElementById('results').style.display = 'none';
   document.getElementById('place').innerHTML = '';
 }
+
+function addSearchBoxToMap(map, input) {
+  // Create the search box and link it to the UI element.
+  const searchBox = new window.google.maps.places.SearchBox(input);
+  map.controls[window.google.maps.ControlPosition.TOP_LEFT].push(input);
+  // Bias the SearchBox results towards current map's viewport.
+  map.addListener("bounds_changed", () => {
+    searchBox.setBounds(map.getBounds());
+  });
+  let markers = [];
+  // Listen for the event fired when the user selects a prediction and retrieve
+  // more details for that place.
+  searchBox.addListener("places_changed", () => {
+    const places = searchBox.getPlaces();
+
+    if (places.length == 0) {
+      return;
+    }
+    // Clear out the old markers.
+    markers.forEach((marker) => {
+      marker.setMap(null);
+    });
+    markers = [];
+    // For each place, get the name and location.
+    const bounds = new window.google.maps.LatLngBounds();
+    places.forEach((place) => {
+      if (!place.geometry) {
+        return;
+      }
+      localStorage.setItem('userLocation',
+          JSON.stringify({lat: place.geometry.location.lat(), lng: place.geometry.location.lng()}));
+      // Create a marker for each place.
+      markers.push(
+        new window.google.maps.Marker({
+          map,
+          title: place.name,
```

Comment:
I think we will only use the marker on this map to mark the user's entered location, so why do we need place.name?

Reply:
This is the marker of the location the user enters in the search box. The title in only so that the name of the place will appear if he presses the marker.

Reply:
But the location is his home adress, not some buisness or known place, so what is the name of these kind of places? I don't see where it's set.

Reply:
Again, this is not the user's location, it's the place that the google search returns (according to the user's input to the search box), and they have attributes like name etc :)

Reply:
Yes I understood that, but since the user's input is usually its home address and not a known place, I was wondering what name is in this attribute, and what kind of name we want it to have. We can deal with this when we work more on the UX, it's just something to pay attention to.

Reply:
It's just the name that comes from google maps, for example a string of his address